### ISSUE TITLE
Command line: SHK/SDK/BXY to 2MG/PO/DSK?

### ISSUE THREAD

**Issue by webspacecreations** [fadden/nulib2#11]
Silly question: On Linux is there a command line incantation that would take a ShrinkIt archive and extract it into a disk image format suitable for loading into an emulator? It looks like nulib2 has the tools, but in reviewing documentation I don't see a clear example of how to accomplish something like this.

**Comment by fadden** [comment]
NuLib2 can extract compressed disk images as .po, and individual files as files, but can't generate a new disk image from a collection of files (which is what it sounds like you're trying to do).  It doesn't understand DOS/ProDOS filesystems.  imgconv converts between .SDK and .2MG, doing more or less what NuLib2 does but adding or stripping the 2MG header.

There's a very crude program called "makedisk" that comes with the CiderPress distribution, in the linux directory (see [README-linux](https://github.com/fadden/ciderpress/blob/master/README-linux.md)).  With it, you can do something like:

    makedisk prodos 800k foo.po file1.txt file2.txt

That would create an 800K ProDOS disk image with two text files on it.  The program is more proof-of-concept than actual utility, and doesn't make any attempt to figure out the file type (which could be a real problem on DOS, where start/len are embedded in 'B' files).

[AppleCommander](https://applecommander.github.io/) has a command-line interface, and might be better suited for your needs.  I haven't tried it myself.

**Comment by webspacecreations** [comment]
> NuLib2 can extract compressed disk images as .po, and individual files as files, but can't generate a new disk image from a collection of files (which is what it sounds like you're trying to do).

Yes, that's what I was hoping to be able to do and wanted to make sure I wasn't overlooking an existing option.

> AppleCommander has a command-line interface, and might be better suited for your needs

Unless more command-line functionality has been added since making the suggestion, AppleCommander really just provides a switch statement to pass images along to the software at startup.

**Thanks for responding with a definitive answer.** My objective is to be able to double-click a ShrinkIt archived file in Linux, have it natively decompressed and piped into a disk image that would then be passed along as the configuration value to an emulator. This is all taking place on a Pi, so doing the decompression in Linux makes a difference vs decompressing in GS/OS. I suspect there's enough info within the AppleCommander and NuLib2 code samples to cobble a solution together; didn't want to start this if it was just reinventing the wheel.
